crates/router/src/core/payment_methods/vault.rs

```diff
@@ -374,6 +377,47 @@ impl Vault {
 }
 
 //------------------------------------------------TokenizeService------------------------------------------------
+pub fn get_key_id(keys: &Jwekey) -> &str {
+    let key_identifier = "1"; // [#46]: Fetch this value from redis or external sources
+    if key_identifier == "1" {
+        &keys.locker_key_identifier1
+    } else {
+        &keys.locker_key_identifier2
+    }
+}
+
+#[cfg(feature = "basilisk")]
+pub async fn get_locker_jwe_keys(
+    keys: &Jwekey,
+) -> CustomResult<(String, String), errors::EncryptionError> {
+    let key_id = get_key_id(keys);
+    if key_id == keys.locker_key_identifier1 {
+        let public_key = services::KeyHandler::get_kms_decrypted_key(
+            keys,
+            keys.locker_encryption_key1.to_string(),
+        )
+        .await?;
+        let private_key = services::KeyHandler::get_kms_decrypted_key(
+            keys,
+            keys.locker_decryption_key1.to_string(),
+        )
+        .await?;
+        Ok((public_key, private_key))
+    } else {
```

Comment:
we need to check `locker_key_identifier2` also equal to key_id, if that fails we need to throw proper error and metrics too.

Reply:
sure